### ISSUE TITLE
Lint/AssignmentInCondition -> Style/RedundantParentheses

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15356]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Lint/AssignmentInCondition

### Steps to reproduce the problem

```ruby
if !a || (
  b = c
  b == 'd'
)
  puts 1
end

# Use `==` if you meant to do a comparison or wrap the expression in parentheses to indicate you meant to assign in a condition. [[Correctable] Lint/AssignmentInCondition]

if !a || (
  (b = c)
  b == 'd'
)
  puts 1
end

# Don't use parentheses around an assignment. [[Correctable] Style/RedundantParentheses]
```

### Expected behavior

A) Accept `b = c`
B) Accept `(b = c)`

B seems surefire, however leaning towards A too, because:

1. Mistyping `==` as `=` pertains to sloppy 90's C coders and their [dumbass rebuttal strategies](https://en.wikipedia.org/wiki/Yoda_conditions)
2. In syntax like this one with multiple statements on multiple lines the mistake is even less probable

Some AST ace will undoubtedly provide insight to complicate the matter.

<img width="224" height="192" alt="Image" src="https://github.com/user-attachments/assets/c04115da-2515-4120-880b-d513ce9dbd10" />

### Actual behavior

Copes gives a suggestion that leads to another cop.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
System Version: macOS 26.5.1 (25F80)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-06-27 1925c07ce
```

**Comment by forthrin** [comment]
Quick shot, Superintendent. Side question: Since labeled a "loop" issue:

1. Does RC detect and silently quit on re-encountering a cop (eg. `warn unless set.add?(cop)`, and shouldn't this throw an exception or notice the user that this is an elevated issue (though not as severe as a parsing error)?
2. Do -- or would -- RC test suites have features to detect loops?
